crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -45,7 +51,191 @@ impl_downcast!(Stage);
 ///
 /// The checker may report a system more times than the amount of constraints it would actually need
 /// to have unambiguous order with regards to a group of already-constrained systems.
-pub struct ReportExecutionOrderAmbiguities;
+///
+/// By default only a warning with the number of unresolved ambiguities detected will be reported per [`SystemStage`].
+/// This behavior can be changed by explicitly adding this resource using the following constructors:
+/// * [ReportExecutionOrderAmbiguities::off()] - Disables all messages reported by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::minimal()] - Displays only the number of unresolved ambiguities detected by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::verbose()] - Displays a full report of ambiguities detected by the ambiguity checker.
+///
+/// The ambiguity checker will ignore ambiguities within official Bevy crates.
+/// To ignore a custom crate, use [`ReportExecutionOrderAmbiguities::ignore`]
+/// with an list of crate names as an argument.
+///
+/// ## Example
+/// ```ignore
+/// # use bevy_app::App;
+/// # use bevy_ecs::schedule::ReportExecutionOrderAmbiguities;
+/// App::new()
+///    .insert_resource(ReportExecutionOrderAmbiguities::verbose().ignore(&["my_external_crate"]));
+/// ```
+pub struct ReportExecutionOrderAmbiguities {
+    pub level: AmbiguityReportLevel,
+    pub ignore_crates: Vec<String>,
+}
+
+const BEVY_CRATES: [&str; 110] = [
```

Comment:
Even though it shouldn't change much, this list seems a good candidate to something that will never be updated and fall out of sync...

Reply:
I fear the same, but I wasn't able to think in a better way of maintaining a list of internal bevy crates.

Reply:
At least it should be documented here: https://github.com/bevyengine/bevy/blob/main/CONTRIBUTING.md?plain=1#L259-L262

Reply:
Done in https://github.com/bevyengine/bevy/pull/2766/commits/889a263f09835e8faecf75805e0f05256cf133ba